### PR TITLE
Update linter config and fix reported issues

### PR DESCRIPTION
**Pull request by fasmat**
This applies the same changes as in https://github.com/spacemeshos/go-spacemesh/pull/3800 to `poet`.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -56,207 +80,487 @@ output:
...
+    #   - stdmethods
+    #   - stringintconv
+    #   - structtag
+    #   - testinggoroutine
+    #   - tests
+    #   - unmarshal
+    #   - unreachable
+    #   - unsafeptr
+    #   - unusedresult
+    #   - unusedwrite
```
Comment (poszu) [inline .golangci.yml]: This list is noise only...  Available analyzers can be found as stated in the comment above:
>Run `go tool vet help` to see all analyzers.

File: .golangci.yml [hunk 2]
```diff
@@ -56,207 +80,487 @@ output:
...
```
Comment (poszu) [inline .golangci.yml]: ditto

File: .golangci.yml [hunk 3]
```diff
@@ -56,207 +80,487 @@ output:
...
   staticcheck:
-    # Select the Go version to target. The default is '1.13'.
-    go: "1.19"
-    # https://staticcheck.io/docs/options#checks
+    # Select the Go version to target.
+    # Default: "1.13"
+    # Deprecated: use the global `run.go` instead.
+    # go: "1.15"
+    # SAxxxx checks in https://staticcheck.io/docs/configuration/options/#checks
+    # Default: ["*"]
```
Comment (poszu) [inline .golangci.yml]: Is it really still 1.13?

**Review by poszu — COMMENTED** [review]
IMO, the .golangci.yml became too noisy. It doesn't make sense to document every field default value, etc., as they _can change_ when the tool is updated. This information should be looked up in the docs if one is interested. Otherwise, it will get out of date pretty soon.

File: .golangci.yml (hunk 3 above)
Comment (fasmat) [reply .golangci.yml]: This file was generated from the default config file for `golangci-lint` so I assume this is correct

File: .golangci.yml (hunk 2 above)
Comment (fasmat) [reply .golangci.yml]: Removed default configuration options that can be found via calling a tool or from a website

File: .golangci.yml (hunk 1 above)
Comment (fasmat) [reply .golangci.yml]: Removed default configuration options that can be found via calling a tool or from a website

**Review by poszu — APPROVED** [review]

**Pull request merged**